static/js/util/util_test.js

```diff
@@ -82,6 +83,41 @@ describe("util", () => {
     })
   })
 
+  describe("getPaymentWithFulfilledOrder", () => {
+    it("gets the payment for the fulfilled order", () => {
+      const klasses = generateFakeKlasses(1, { hasPayment: true })
+      const klass = klasses[0]
+      assert.deepEqual(
+        getPaymentWithFulfilledOrder(
+          klass.payments,
+          klass.payments[0].order.id
+        ),
+        klass.payments[0]
+      )
+    })
+
+    it("returns undefined if there are no payments", () => {
+      const klasses = generateFakeKlasses(1)
+      assert.deepEqual(
+        getPaymentWithFulfilledOrder(klasses[0].payments, 1),
+        undefined
+      )
+    })
+
+    it("returns undefined if the order is not fulfilled", () => {
+      const klasses = generateFakeKlasses(1, { hasPayment: true })
+      const klass = klasses[0]
+      klass.payments[0].order.status = "created"
+      assert.deepEqual(
+        getPaymentWithFulfilledOrder(
+          klass.payments,
+          klass.payments[0].order.id
+        ),
+        undefined
+      )
+    })
```

Comment:
I think this test case and the first test case at line 87 can be parametrized. I don't actually know if this will work & I'm not wild about the variable scope & the changing of value inside a `beforeEach`. Anyway, I think it's valuable to parametrize very similar tests where we can. Let me know what you think

```javascript
const klass, payment;

beforeEach(() => {
  klass = generateFakeKlasses(1, { hasPayment: true })[0]
  payment = klass.payments[0]
})

[
  ["created", undefined, "undefined", "not fulfilled"],
  ["fulfilled", payment, "a payment", "fulfilled"],
].forEach((orderStatus, expectedValue, valueDescriptor, statusDescriptor) => {
  it(`returns ${valueDescriptor} if the order is {statusDescriptor}`, () => {
      klass.payments[0].order.status = orderStatus
      assert.deepEqual(
        getPaymentWithFulfilledOrder(
          klass.payments,
          klass.payments[0].order.id
        ),
        expectedValue
      )
    })
```

Reply:
In this case I feel like parameterization would make it more difficult to understand at a glance what the test code is trying to accomplish; it would sacrifice readability for compactness.  But I like the idea of creating the klass and payment in `beforeEach`

Reply:
Added the `beforeEach` function but not parameters here and in previous test case, let me know if okay